src/include/error.h

```diff
@@ -41,77 +45,111 @@
 #define __wt_set_return(session, error) \
     __wt_set_return_func(session, __PRETTY_FUNCTION__, __LINE__, error, #error)
 
+#ifdef WT_HAVE_ERROR_LOG
+#define __wt_error_log_add_helper(expr, error, suberror) \
+    __wt_error_log_add(__FILE__, __PRETTY_FUNCTION__, __LINE__, expr, error, suberror)
+#else
+#define __wt_error_log_add_helper(expr, error, suberror)
+#endif
+
+/*
+ * WT_ERROR_LOG_ADD --
+ *     Add an entry to the error log, regardless of whether it is enabled, and return the error
+ *     code. This is useful for important error codes that should always be logged. However, using
+ *     this macro currently results in the error being logged twice if the error log is enabled.
+ */
+#define WT_ERROR_LOG_ADD(expr) \
+    __wt_error_log_add_ret(__FILE__, __PRETTY_FUNCTION__, __LINE__, #expr, expr, WT_NONE)
+
 /* Set "ret" and branch-to-err-label tests. */
-#define WT_ERR(a)             \
-    do {                      \
-        if ((ret = (a)) != 0) \
-            goto err;         \
+#define WT_ERR(a)                                        \
+    do {                                                 \
+        if ((ret = (a)) != 0) {                          \
+            __wt_error_log_add_helper(#a, ret, WT_NONE); \
+            goto err;                                    \
+        }                                                \
     } while (0)
 #define WT_ERR_MSG(session, v, ...)                                    \
     do {                                                               \
         ret = (v);                                                     \
+        __wt_error_log_add_helper(#v, ret, WT_NONE);                   \
         __wt_err(session, ret, __VA_ARGS__);                           \
         __wt_session_set_last_error(session, v, WT_NONE, __VA_ARGS__); \
         goto err;                                                      \
     } while (0)
 #define WT_ERR_SUB(session, v, sub_v, ...)                           \
     do {                                                             \
         ret = (v);                                                   \
+        __wt_error_log_add_helper(#v, ret, sub_v);                   \
         __wt_session_set_last_error(session, v, sub_v, __VA_ARGS__); \
         goto err;                                                    \
     } while (0)
+#define WT_ERR_TEST(a, v, keep)                          \
+    do {                                                 \
+        if (a) {                                         \
+            ret = (v);                                   \
+            __wt_error_log_add_helper(#v, ret, WT_NONE); \
+            goto err;                                    \
+        } else if (!(keep))                              \
+            ret = 0;                                     \
```

Comment:
Would it make sense to also clear the errors here?

Additional improvement suggestion (optional?):

1.
Add a macro like "RESET_RET()" like this:
```c
#define RESET_RET() do { __wt_error_log_clear(); ret = 0; } while(0)
```

2.
Find all occurrences of "  ret = 0;" in the code and replace them with `RESET_RET()`